cmake/modules/extensions.cmake

```diff
@@ -1269,8 +1269,16 @@ function(zephyr_linker_sources location)
   elseif("${location}" STREQUAL "NOCACHE_SECTION")
     set(snippet_path "${nocache_path}")
   elseif("${location}" STREQUAL "ITCM_SECTION")
+    dt_has_chosen(HAS_ITCM PROPERTY "zephyr,itcm")
+    if(NOT HAS_ITCM)
+      message(FATAL_ERROR "Trying to link snippet into itcm but no itcm available.")
```

Comment:
minor, but would it be possible to provide a little extra info to the user when this happen, as to how to address ?
For example on adding `itcm` to devicetree / use a device which support itcm.

Reply:
done